source/extensions/filters/network/redis_proxy/router_impl.cc

```diff
@@ -87,17 +87,19 @@ RouteSharedPtr PrefixRoutes::upstreamPool(std::string& key) {
     value = prefix_lookup_table_.findLongestPrefix(key.c_str());
   }
 
+  if (value == nullptr) {
+    // prefix route not found, default to catch all route.
+    value = catch_all_route_;
+  }
   if (value != nullptr) {
```

Comment:
Do we need this check any more?

Reply:
it breaks a unit test if `catch_all_route` is not defined.

Reply:
Is n't one of routes/catch-all-route is mandatory? If no prefix routes are specified, catch-all-route becomes mandatory?

Reply:
this test breaks (`MissingCatchAll`) since longest prefix matching route is not found and `catchAllRoute` is not configured in this case. 

https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/network/redis_proxy/router_impl_test.cc#L52-L63 ,

Reply:
So isn't that a test bug? How do we get to this case if it can't pass config validation?

/wait

Reply:
@mattklein123 - We can't get to this case without config validation. So upstreamPool should never return `nullptr`. However in this case its a unit test. I validated this locally. 

Would it make sense to remove the unit test and simplify the code flow without needing the extra `nullptr` check given config validation will anyways prevent proxy from starting up ?

Reply:
Yes if the code can't actually be hit I would fix the test.